### DISCUSSION TITLE
Metamask Javascript SDK in ionic not working in iOS

### DISCUSSION THREAD

**Discussion by alihaider1793** [MetaMask/metamask-sdk#1110] [Q&A]
Hi, I am using metamask/sdk in my ionic app for connecting metamask wallet with my app
I want to get the wallet address when the user clicks on the connect button within the app
It's working in android devices but when I try to test this in iOS devices, it's not working. It opens the metamask wallet and displays a modal that says "connecting to metamask" and then nothing happens.

How can I fix this?

**Comment by FlaviuWeb3** [comment]
hi @alihaider1793 Would you please share with me your dependencies versions from package.json?

**Reply by alihaider1793** [reply]
`{
  "name": "DemoProject",
  "version": "0.0.1",
  "author": "Ionic Framework",
  "homepage": "https://ionicframework.com/",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~13.2.2",
    "@angular/core": "~13.2.2",
    "@angular/forms": "~13.2.2",
    "@angular/platform-browser": "~13.2.2",
    "@angular/platform-browser-dynamic": "~13.2.2",
    "@angular/router": "~13.2.2",
    "@awesome-cordova-plugins/clipboard": "5.44.0",
    "@awesome-cordova-plugins/core": "5.41.0",
    "@awesome-cordova-plugins/in-app-browser": "^6.0.1",
    "@capacitor-community/stripe": "^4.1.2",
    "@capacitor/android": "3.6.0",
    "@capacitor/app": "1.1.1",
    "@capacitor/app-launcher": "1.0.9",
    "@capacitor/core": "3.6.0",
    "@capacitor/haptics": "1.1.4",
    "@capacitor/ios": "3.6.0",
    "@capacitor/keyboard": "1.2.3",
    "@capacitor/network": "1.0.7",
    "@capacitor/splash-screen": "^1.2.2",
    "@capacitor/status-bar": "1.0.8",
    "@ionic-native/base64": "^5.36.0",
    "@ionic-native/clipboard": "5.36.0",
    "@ionic-native/core": "5.36.0",
    "@ionic-native/nfc": "^5.36.0",
    "@ionic/angular": "6.0.0",
    "@ionic/cordova-builders": "6.1.0",
    "@ionic/storage-angular": "3.0.6",
    "@metamask/sdk": "^0.14.3",
    "@stripe/stripe-js": "^3.3.0",
    "@techiediaries/ngx-qrcode": "9.1.0",
    "@walletconnect/ethereum-provider": "^2.9.2",
    "@walletconnect/modal": "^2.6.1",
    "@web3auth/web3auth": "1.1.0",
    "assert": "2.0.0",
    "browserify-zlib": "^0.2.0",
    "buffer": "6.0.3",
    "cordova-clipboard": "1.3.0",
    "cordova-plugin-inappbrowser": "^5.0.0",
    "crypto-browserify": "3.12.0",
    "https-browserify": "1.0.0",
    "ionic4-emoji-picker": "0.0.2",
    "jetifier": "^2.0.0",
    "magic-sdk": "8.1.1",
    "moralis-v1": "^1.13.0",
    "next-tick": "^1.1.0",
    "ng-otp-input": "1.8.5",
    "os-browserify": "0.3.0",
    "parse": "^4.2.0",
    "path-browserify": "^1.0.1",
    "phonegap-nfc": "^1.2.0",
    "process": "0.11.10",
    "rxjs": "~6.6.0",
    "stream-browserify": "3.0.0",
    "stream-http": "3.2.0",
    "tslib": "2.2.0",
    "url": "0.11.0",
    "web3": "^1.7.4",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~13.2.3",
    "@angular-eslint/builder": "~13.0.1",
    "@angular-eslint/eslint-plugin": "~13.0.1",
    "@angular-eslint/eslint-plugin-template": "~13.0.1",
    "@angular-eslint/template-parser": "~13.0.1",
    "@angular/cli": "~13.2.3",
    "@angular/compiler": "~13.2.2",
    "@angular/compiler-cli": "~13.2.2",
    "@angular/language-service": "~13.2.2",
    "@capacitor/cli": "3.6.0",
    "@ionic/angular-toolkit": "6.0.0",
    "@ionic/lab": "3.2.11",
    "@types/jasmine": "~3.6.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "12.11.1",
    "@typescript-eslint/eslint-plugin": "5.3.0",
    "@typescript-eslint/parser": "5.3.0",
    "cordova-android": "10.1.2",
    "cordova-plugin-androidx-adapter": "1.1.3",
    "cordova-plugin-statusbar": "2.4.2",
    "eslint": "7.6.0",
    "eslint-plugin-import": "2.22.1",
    "eslint-plugin-jsdoc": "30.7.6",
    "eslint-plugin-prefer-arrow": "1.2.2",
    "jasmine-core": "~3.8.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~6.3.2",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "1.5.0",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "typescript": "~4.4.4"
  },
  "browser": {
    "fs": false,
    "net": false,
    "tls": false,
    "child_process": false
  },
  "android": {
    "fs": false,
    "net": false,
    "tls": false,
    "child_process": false
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-statusbar": {},
      "cordova-plugin-device": {},
      "cordova-plugin-ionic-webview": {},
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-androidx-adapter": {},
      "cordova-clipboard": {}
    },
    "platforms": [
      "android"
    ]
  }
}
`

**Comment by christopherferreira9** [comment]
Hi @alihaider1793 ! Can you share the MetaMask options you're using or provide a repo we can check the implementation on?
